### PR TITLE
Adding missing datasource singleton annoataion

### PR DESCRIPTION
**Pull request by kunkun-tang**
Both `H2FileDataSource ` and `MySQLDataSource ` should be singleton. The reason behind is that We would like all database operations to share the same connection pool to guarantee performance. 

Besides, removing singleton from `DatabaseOperator` as no states exist there. It is safe to have multiple DatabaseOperator instances.

### CODE REVIEW

**Comment by HappyRay** [comment]
Do they need to be Singleton?
Is there a problem the way they are?
Singleton should be a special case, not a norm.

**Comment by kunkun-tang** [comment]
@HappyRay I don't think we necessarily create multiple data source instances. One datasource should be able to manage all connections. Actually, I just realized it's already a singleton, since DatabaseOperator is set to singleton, and it is the only place to call AzkabanDataSource.

https://github.com/azkaban/azkaban/blob/b4d893fd7fd02fe50c090a1de55e6baf26ee5108/azkaban-db/src/main/java/azkaban/db/DatabaseOperator.java#L37-L38

**Comment by kunkun-tang** [comment]
Updated description.

File: azkaban-web-server/src/test/java/azkaban/webapp/AzkabanWebServerTest.java [hunk 1]
```diff
@@ -147,8 +148,7 @@ public void testInjection() throws Exception {
     assertSingleton(ExecutorEventsDao.class, injector);
     assertSingleton(ActiveExecutingFlowsDao.class, injector);
     assertSingleton(FetchActiveFlowDao.class, injector);
-    assertSingleton(AzkabanWebServer.class, injector);
-
+    assertSingleton(H2FileDataSource.class, injector);
```
Comment (HappyRay) [inline azkaban-web-server/src/test/java/azkaban/webapp/AzkabanWebServerTest.java]: How about the mysql datasource?

File: azkaban-web-server/src/test/java/azkaban/webapp/AzkabanWebServerTest.java [hunk 2]
```diff
@@ -147,8 +148,7 @@ public void testInjection() throws Exception {
     assertSingleton(ExecutorEventsDao.class, injector);
     assertSingleton(ActiveExecutingFlowsDao.class, injector);
     assertSingleton(FetchActiveFlowDao.class, injector);
-    assertSingleton(AzkabanWebServer.class, injector);
```
Comment (HappyRay) [inline azkaban-web-server/src/test/java/azkaban/webapp/AzkabanWebServerTest.java]: Why this change?

File: azkaban-web-server/src/test/java/azkaban/webapp/AzkabanWebServerTest.java (hunk 1 above)
Comment (kunkun-tang) [reply azkaban-web-server/src/test/java/azkaban/webapp/AzkabanWebServerTest.java]: Mysql dataSource cannot be added here, since Guice only have h2 Props, which has conflicts against Mysql Props.

File: azkaban-web-server/src/test/java/azkaban/webapp/AzkabanWebServerTest.java (hunk 2 above)
Comment (kunkun-tang) [reply azkaban-web-server/src/test/java/azkaban/webapp/AzkabanWebServerTest.java]: This should be a typo. Adding back.

File: azkaban-web-server/src/test/java/azkaban/webapp/AzkabanWebServerTest.java (hunk 1 above)
Comment (HappyRay) [reply azkaban-web-server/src/test/java/azkaban/webapp/AzkabanWebServerTest.java]: Can it be tested separately? It can be in a separate PR.
